### PR TITLE
Update Go Components page - to align with 2.1.0

### PR DESCRIPTION
**Pull request by tpmccallum**
Signed-off-by: tpmccallum tim.mccallum@fermyon.com

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors
- [x] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

**Comment by tpmccallum** [comment]
References: 
https://github.com/golang/go/issues/31105#issuecomment-1552362280
https://tip.golang.org/doc/go1.21

File: content/spin/v2/go-components.md [hunk 1]
```diff
@@ -157,12 +184,9 @@ now receive requests in route `/hello`:
 <!-- @selectiveCpy -->
 
 ```bash
-$ curl -i localhost:3000/hello
+$ curl -i localhost:3000
```
Comment (itowlson) [inline content/spin/v2/go-components.md:187]: This no longer matches the manifest (I don't know whether you want to change the route in the manifest or restore it here, but if the former then line 182 will also need a change).

Comment (tpmccallum) [reply content/spin/v2/go-components.md:187]: Great catch, thank you.
I have updated the `curl` command to make the request to the correct endpoint.

File: content/spin/v2/go-components.md [hunk 2]
```diff
@@ -67,6 +65,35 @@ func init() {
...
 
+The Spin HTTP component (written in Go) can be built using the `tingygo` toolchain:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ tinygo build -o main.wasm -target=wasi main.go
+```
+
+Once built, we can run our Spin HTTP component (written in Go) using the Spin up command:
```
Comment (itowlson) [inline content/spin/v2/go-components.md:76]: nit: we don't need to repeat "written in Go" so soon

```suggestion
Once built, we can run our Spin HTTP component using the Spin up command:
```

File: content/spin/v2/go-components.md [hunk 3]
```diff
@@ -67,6 +65,35 @@ func init() {
...
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin up
+```
+
+The Spin HTTP component (written in Go) can now receive and process incoming requests:
```
Comment (itowlson) [inline content/spin/v2/go-components.md:84]:
```suggestion
The Spin HTTP component can now receive and process incoming requests:
```

(and are you comfortable with the multiple repetition of "Spin HTTP component"?  It starts to feel a little awkward to me but maybe that's a matter of taste)

**Review by itowlson — APPROVED** [review]
Little nit that I missed before (sorry) but LGTM!

**Pull request merged**